tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,268 @@
+                .withRequiredArg()
+                .describedAs("prop")
+                .ofType(String.class);
+        messageFormatterConfigOpt = parser.accepts("formatter-config", "Config properties file to initialize the message formatter. Note that " + messageFormatterArgOpt + " takes precedence over this config.")
+                .withRequiredArg()
+                .describedAs("config file")
+                .ofType(String.class);
+        maxMessagesOpt = parser.accepts("max-messages", "The maximum number of messages to consume before exiting. If not set, consumption is continual.")
+                .withRequiredArg()
+                .describedAs("num_messages")
+                .ofType(Integer.class);
+        timeoutMsOpt = parser.accepts("timeout-ms", "If specified, exit if no message is available for consumption for the specified interval.")
+                .withRequiredArg()
+                .describedAs("timeout_ms")
+                .ofType(Integer.class);
+        rejectOpt = parser.accepts("reject", "If specified, messages are rejected as they are consumed.");
+        releaseOpt = parser.accepts("release", "If specified, messages are released as they are consumed.");
+        rejectMessageOnErrorOpt = parser.accepts("reject-message-on-error", "If there is an error when processing a message, " +
+                "reject it instead of halt.");
+        bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED: The server(s) to connect to.")
+                .withRequiredArg()
+                .describedAs("server to connect to")
+                .ofType(String.class);
+        keyDeserializerOpt = parser.accepts("key-deserializer")
+                .withRequiredArg()
+                .describedAs("deserializer for key")
+                .ofType(String.class);
+        valueDeserializerOpt = parser.accepts("value-deserializer")
+                .withRequiredArg()
+                .describedAs("deserializer for values")
+                .ofType(String.class);
+        groupIdOpt = parser.accepts("group", "The share group id of the consumer.")
+                .withRequiredArg()
+                .describedAs("share group id")
+                .ofType(String.class);
+
+        try {
+            options = parser.parse(args);
+        } catch (OptionException oe) {
+            CommandLineUtils.printUsageAndExit(parser, oe.getMessage());
+        }
+
+        CommandLineUtils.maybePrintHelpOrVersion(this, "This tool helps to read data from Kafka topics using share groups and outputs it to standard output.");
+
+        checkRequiredArgs();
+
+        if (options.has(rejectOpt) && options.has(releaseOpt)) {
+            CommandLineUtils.printUsageAndExit(parser, "At most one of --reject and --release may be specified.");
+        }
```

Comment:
I don't find a test for this, can we please add.

Reply:
Okay, I have added now.